src/win/util.c

```diff
@@ -1665,17 +1665,16 @@ int uv_os_unsetenv(const char* name) {
 
 int uv_os_gethostname(char* buffer, size_t* size) {
   WCHAR buf[UV_MAXHOSTNAMESIZE];
+  DWORD buf_len = UV_MAXHOSTNAMESIZE;
   size_t len;
   char* utf8_str;
   int convert_result;
 
   if (buffer == NULL || size == NULL || *size == 0)
     return UV_EINVAL;
 
-  uv__once_init(); /* Initialize winsock */
-
-  if (GetHostNameW(buf, UV_MAXHOSTNAMESIZE) != 0)
-    return uv_translate_sys_error(WSAGetLastError());
+  if (GetComputerNameExW(ComputerNameDnsFullyQualified, buf, &buf_len) == 0)
```

Comment:
Do we know if this should be an FQDN, or perhaps the truncated NetBIOS name, or just the DNS hostname?
 ```suggestion
  if (GetComputerNameExW(ComputerNameNetBIOS, buf, &buf_len) == 0)
```

Reply:
ComputerNameDnsHostname can be a good alternative too. There is a more detailed explanation in the separate post.